File: frontend/src/services/admin/reviews.ts
import { apiClient } from '../apiClient'
import type { Envelope } from '@/types/catalog'
import type { AdminReview, AdminReviewListResponse } from '@/types/review'

export async function listAdminReviews(
  params: { status?: string; page?: number; per_page?: number } = {},
): Promise<AdminReviewListResponse> {
  const { data } = await apiClient.get<AdminReviewListResponse>('/admin/reviews', { params })
  return data
}

export async function moderateReview(
  reviewId: number,
  status: 'approved' | 'rejected' | 'hidden',
): Promise<AdminReview> {
  const { data } = await apiClient.post<Envelope<AdminReview>>(`/admin/reviews/${reviewId}/moderate`, { status })
  return data.data
}

File: backend/app/api/v1/admin/reviews.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_db, require
from app.models.auth import User
from app.schemas.review import AdminReviewListOut, AdminReviewModerateIn, AdminReviewOut
from app.services.review_service import AdminReviewService

router = APIRouter(prefix="/admin/reviews", tags=["admin-reviews"])


@router.get("", response_model=AdminReviewListOut)
async def list_reviews(
    db: AsyncSession = Depends(get_db),
    _user: User = Depends(require("review:review:moderate")),
    status: str | None = Query(default=None),
    page: int = Query(default=1, ge=1),
    per_page: int = Query(default=20, ge=1, le=100),
):
    return await AdminReviewService(db).list_for_moderation(status=status, page=page, per_page=per_page)


@router.post("/{review_id}/moderate", response_model=AdminReviewOut)
async def moderate_review(
    review_id: int,
    payload: AdminReviewModerateIn,
    db: AsyncSession = Depends(get_db),
    user: User = Depends(require("review:review:moderate")),
):
    return await AdminReviewService(db).moderate(review_id, user.id, payload.status)
